app/models/program_session.rb

```diff
@@ -9,12 +9,33 @@ class ProgramSession < ActiveRecord::Base
   belongs_to :track
   belongs_to :session_format
   has_one :time_slot
+  has_many :speakers
 
   validates :event, :session_format, :title, :state, presence: true
 
   scope :unscheduled, -> do
     where(state: ACTIVE).where.not(id: TimeSlot.pluck(:program_session_id))
   end
+
+  def self.create_from_proposal(proposal)
+    self.transaction do
+      ps = ProgramSession.create!(event_id: proposal.event_id,
+                                  proposal_id: proposal.id,
+                                  title: proposal.title,
+                                  abstract: proposal.abstract,
+                                  track_id: proposal.track_id,
+                                  session_format_id: proposal.session_format_id)
+
+      #attach proposal speakers to new program session
+      ps.speakers << proposal.speakers
+      ps.speakers.each do |speaker|
```

Comment:
@gitcow you were right, the speakers weren't actually being attached to the program session and saved in the db. Good catch. I'm not sure if this is the best way to handle this - seems un-elegant and like a novice move - but it seems to be working properly now.